packages/account/src/Sections/Profile/PersonalDetails/verify-button.tsx

```diff
@@ -8,94 +8,105 @@ import { useTranslations, Localize } from '@deriv-com/translations';
 import { useDevice } from '@deriv-com/ui';
 import './verify-button.scss';
 
+type TSetStatus = {
+    msg: string;
+    code: string;
+};
+
 type TVerifyButton = {
     is_verify_button_disabled: boolean;
     next_email_otp_request_timer?: number;
+    setStatus: ({ msg, code }: TSetStatus) => void;
 };
 
-export const VerifyButton = observer(({ is_verify_button_disabled, next_email_otp_request_timer }: TVerifyButton) => {
-    const { client, ui } = useStore();
-    const { setShouldShowPhoneNumberOTP, is_scroll_to_verify_button, setIsScrollToVerifyButton } = ui;
-    const { account_settings, setVerificationCode } = client;
-    const { phone_number_verification } = account_settings;
-    const is_phone_number_verified = !!phone_number_verification?.verified;
-    const history = useHistory();
-    const { localize } = useTranslations();
-    //@ts-expect-error remove this when phone_number_verification is added to api calls
-    const { sendPhoneNumberVerifyEmail, WS, is_loading } = useVerifyEmail('phone_number_verification');
-    const { isDesktop } = useDevice();
-    const ref = useRef<HTMLDivElement | null>(null);
+export const VerifyButton = observer(
+    ({ is_verify_button_disabled, next_email_otp_request_timer, setStatus }: TVerifyButton) => {
+        const { client, ui } = useStore();
+        const { setShouldShowPhoneNumberOTP, is_scroll_to_verify_button, setIsScrollToVerifyButton } = ui;
+        const { account_settings, setVerificationCode } = client;
+        const { phone_number_verification } = account_settings;
+        const is_phone_number_verified = !!phone_number_verification?.verified;
```

Comment:
@suisin-deriv just noticed that we are checking this condition in multiple places. should we create a hook `useIsphoneNumberVerified` instead?